libmscore/edit.cpp

```diff
@@ -2162,7 +2162,7 @@ void Score::deleteAnnotationsFromRange(Segment* s1, Segment* s2, int track1, int
       for (int track = track1; track < track2; ++track) {
             if (!filter.canSelectVoice(track))
                   continue;
-            for (Segment* s = s1; s != s2; s = s->next1()) {
+            for (Segment* s = s1; s && s != s2; s = s->next1()) {
```

Comment:
Usually `next1()` returns a null pointer when it reaches the end of a score, is it not true in case of this issue? I mean, if you place some other notes with annotations after that MM-rest, wouldn't this lead to these annotations be unnecessarily removed by this function? In my tests it doesn't happen but this means that `next1()` probably returns a null pointer somewhere in the middle of a score.

Reply:
It probably returns a `nullptr` when the multimeasure rest is reached. See the score: https://musescore.org/sites/musescore.org/files/2019-11/Voices%20mmrests%20copy%201.mscz.

Reply:
So were you suggesting that I should make changes to `next1()`?

Reply:
No, `next1()` should be fine, I just tried to understand what is so different about internal structure of segments or measures when the file loads with MM rests vs. when MM rests are regenerated that makes `next1()` work so unexpectedly. The check you have added should be fine, but there seem to be other code that could rely on `next1()` returning `nullptr` only when the score itself ends, so this may lead to other bugs as well.

Reply:
So those code which ought to give `next1()` the expected result will have to be fixed to get the issue resolved?